ale_linters/powershell/psscriptanalyzer.vim

```diff
@@ -0,0 +1,154 @@
+" Author: Jesse Harris - https://github.com/zigford
+" Description: This file adds support for lintng powershell scripts
+"   using the PSScriptAnalyzer module.
+
+" let g:ale_powershell_psscriptanalyzer_exclusions =
+" \ 'PSAvoidUsingWriteHost,PSAvoidGlobalVars'
+call ale#Set('powershell_psscriptanalyzer_exclusions',
+\ get(g:,
+\ 'ale_linters_powershell_psscriptanalyzer_exclusions', ''))
+call ale#Set('powershell_psscriptanalyzer_executable', 'pwsh')
+call ale#Set('powershell_psscriptanalyzer_module',
+\ 'psscriptanalyzer')
+
+" GetExecutable {{{
+function! ale_linters#powershell#psscriptanalyzer#GetExecutable(buffer) abort
+    return ale#Var(a:buffer, 'powershell_psscriptanalyzer_executable')
+endfunction
+" }}}
+
+" PowerShell Escape {{{
+" Powershell variables use $s, so they need to be escaped
+" if being used from a unix shell
+function! ale_linters#powershell#psscriptanalyzer#Escape(str) abort
+    if fnamemodify(&shell, ':t') !~# '\(powershell\|cmd\|pwsh\)'
+        " Powershell commands use $ for variables and must be
+        " escaped in bash
+        return substitute(
+        \   a:str,
+        \   '\$',
+        \   '\\$',
+        \   'g'
+        \)
+    endif
+
+    return a:str
+endfunction
+" }}}
+
+" VersionCheck {{{
+" return the module version
+" Not used yet
+function! ale_linters#powershell#psscriptanalyzer#VersionCheck(buffer) abort
+    let l:executable =
+    \ ale_linters#powershell#psscriptanalyzer#GetExecutable(a:buffer)
+    let l:module = ale#Var(a:buffer, 'powershell_psscriptanalyzer_module')
+
+    if ale#semver#HasVersion(l:module)
+        return ''
+    endif
+
+    "let l:executable = ale#Escape(l:executable)
+    let l:module_string = ale_linters#powershell#psscriptanalyzer#Escape(
+    \ ' -NoProfile -Command "&{$m=Get-Module -ListAvailable '
+    \ . l:module . ';$m.Version.ToString()}"')
+
+    return l:executable . l:module_string
+endfunction
+ " }}}
+
+" RunPowerShell {{{
+" Write a powershell script to a temp file for execution
+" return the command used to execute it
+function! s:TemporaryPSScript(buffer, input) abort
+    let l:filename = 'script.ps1'
+
+    " Create a temporary filename, <temp_dir>/<original_basename>
+    " The file itself will not be created by this function.
+    let l:tempscript =
+    \ ale#util#Tempname() . (has('win32') ? '\' : '/') . l:filename
+
+    if ale#command#CreateTempFile(a:buffer, l:tempscript, a:input)
```

Comment:
Use `ale#command#CreateFile` or `ale#Command#ManageFile` instead. `CreateTempFile` isn't intended for use outside of some very specific parts of code.

Reply:
Thanks for your feedback, implemented your suggestions in commit a839a9e4197101182accdf9cf48867def5479bec